pkg/controllers/service/service_lb_controller.go

```diff
@@ -0,0 +1,155 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package service
+
+import (
+	"context"
+	"os"
+	"runtime"
+
+	v1 "k8s.io/api/core/v1"
+	apimachineryruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/version"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/controllers/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	"github.com/vmware-tanzu/nsx-operator/pkg/metrics"
+	_ "github.com/vmware-tanzu/nsx-operator/pkg/nsx/ratelimiter"
+	servicecommon "github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+var (
+	log           = logger.Log
+	ResultNormal  = common.ResultNormal
+	ResultRequeue = common.ResultRequeue
+	MetricResType = common.MetricResTypeServiceLb
+)
+
+// ServiceLbReconciler reconciles a Service LoadBalancer object
+type ServiceLbReconciler struct {
+	Client   client.Client
+	Scheme   *apimachineryruntime.Scheme
+	Service  *servicecommon.Service
+	Recorder record.EventRecorder
+}
+
+func updateSuccess(r *ServiceLbReconciler, c *context.Context, lbService *v1.Service) {
+	r.setServiceLbStatus(c, lbService)
+	r.Recorder.Event(lbService, v1.EventTypeNormal, common.ReasonSuccessfulUpdate, "LoadBalancer service has been successfully updated")
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateSuccessTotal, MetricResType)
+}
+
+func (r *ServiceLbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	service := &v1.Service{}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, service); err != nil {
+		log.Error(err, "unable to fetch lb service", "req", req.NamespacedName)
+		return ResultNormal, client.IgnoreNotFound(err)
+	}
+
+	if service.Spec.Type == v1.ServiceTypeLoadBalancer {
+		log.Info("reconciling lb service", "lbService", req.NamespacedName)
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, MetricResType)
+
+		if service.ObjectMeta.DeletionTimestamp.IsZero() && r.Service.NSXConfig.EnableVPCNetwork {
```

Comment:
could we check the EnableVPCNetwork before launch the service_lb_controller?

Reply:
Yes, correct. I think we can remove this check from here because we already checked at https://github.com/vmware-tanzu/nsx-operator/pull/541/files#diff-c444f711e9191b53952edb65bfd8c644419fc7695c62611dc0fb304b4fb197d6R242
when starting service_lb_controller